pkg/agent/controller/networkpolicy/priority.go

```diff
@@ -277,19 +277,51 @@ func (pa *priorityAssigner) GetOFPriority(p types.Priority) (*uint16, map[uint16
 		return &ofPriority, nil, nil, nil
 	}
 	insertionPoint, occupied := pa.getInsertionPoint(p)
-	if insertionPoint == PriorityBottomCNP || insertionPoint > PriorityTopCNP || occupied {
+	if insertionPoint == PolicyBottomPriority || insertionPoint > PolicyTopPriority || occupied {
 		return pa.reassignPriorities(insertionPoint, p)
 	}
 	pa.updatePriorityAssignment(insertionPoint, p)
 	return &insertionPoint, nil, nil, nil
 }
 
 // RegisterPriorities registers a list of Priorities to be created with the priorityAssigner.
-// It is used to populate the priorityMap in case of batch rule adds.
+// It is used to populate the priorityMap in case of batch rule adds. Note that this function assumes
+// that the pa.ofPriorityMap is empty at the time when it is called.
 func (pa *priorityAssigner) RegisterPriorities(priorities []types.Priority) error {
-	for _, p := range priorities {
-		if _, _, _, err := pa.GetOFPriority(p); err != nil {
-			return err
+	if len(pa.ofPriorityMap) != 0 {
+		return fmt.Errorf("cannot batch register priorities with a non empty priority map")
```

Comment:
I do think `klog.Fatalf` may be more appropriate error, as it would indicate something wrong with the code, not a transient or user error.

Reply:
We do need to return an error to the calling function though right...? Are you suggesting adding a klog.Fatalf() msg before the return statement?

Reply:
You don't need to return anything

`klog.Fatalf` calls `os.Exit`

now if there is any chance of it being a transient error, this is not the right call; but if it indicates a pretty serious bug in the code, it's better IMO

Reply:
Got it. If that's the case maybe let's not call `klog.Fatalf` here. Worst case scenario, this function is called when the priorityMap is not empty, the batch registration process will error out and the reconciler would fall back to register policy rules individually. It's not something as serious as we should crash Antrea for.

Reply:
makes sense